src/main/java/org/ow2/proactive/procci/model/occi/infrastructure/ComputeBuilder.java

```diff
@@ -97,38 +110,69 @@ public ComputeBuilder(Model cloudAutomation) throws SyntaxException {
+                attributes -> attributes.getOrDefault(ARCHITECTURE_NAME, null))));
+        this.state = getStateFromString(ConvertUtils.getStringFromObject(Optional.ofNullable(
+                rendering.getAttributes()).map(
+                attributes -> attributes.getOrDefault(COMPUTE_STATE_NAME, null))));
+        this.hostname = ConvertUtils.getStringFromObject(Optional.ofNullable(
+                rendering.getAttributes()).map(attributes -> attributes.getOrDefault(HOSTNAME_NAME, null)));
         this.cores = ConvertUtils.getIntegerFromString(
-                Optional.ofNullable(rendering.getAttributes().getOrDefault(CORES_NAME, null))
+                Optional.ofNullable(rendering.getAttributes())
+                        .map(attributes -> attributes.getOrDefault(CORES_NAME, null))
                         .map(coreNumber -> String.valueOf(coreNumber)));
         this.memory = ConvertUtils.getFloatFromString(
-                Optional.ofNullable(rendering.getAttributes().getOrDefault(MEMORY_NAME, null))
+                Optional.ofNullable(rendering.getAttributes())
+                        .map(attributes -> attributes.getOrDefault(MEMORY_NAME, null))
                         .map(memoryNumber -> String.valueOf(memoryNumber)));
         this.share = ConvertUtils.getIntegerFromString(
-                Optional.ofNullable(rendering.getAttributes().getOrDefault(SHARE_NAME, null))
+                Optional.ofNullable(rendering.getAttributes())
+                        .map(attributes -> attributes.getOrDefault(SHARE_NAME, null))
                         .map(shareNumber -> String.valueOf(shareNumber)));
-        this.summary = Optional.ofNullable(
-                (String) rendering.getAttributes().getOrDefault(SUMMARY_NAME, null));
+        this.summary = ConvertUtils.getStringFromObject(Optional.ofNullable(
+                rendering.getAttributes()).map(attributes -> attributes.getOrDefault(SUMMARY_NAME, null)));
         this.mixins = new ArrayList<>();
+        for (String mixin : Optional.ofNullable(rendering.getMixins()).orElse(new ArrayList<>())) {
+            this.mixins.add(providerMixin.getMixinByTitle(mixin));
+        }
+        associateProviderMixin(rendering.getAttributes());
+
         this.links = new ArrayList<>();
+
+        return this;
     }
 
+    void associateProviderMixin(Map<String, Object> attributes) throws ClientException {
+        if (attributes == null) {
+            return;
+        }
+        for (String mixinName : attributes.keySet()) {
+            if (providerMixin.getInstance(mixinName).isPresent()) {
+                try {
+                    this.mixins.add(providerMixin.getInstance(mixinName).get().build(
+                            (Map) attributes.get(mixinName)));
+                } catch (ClassCastException e) {
+                    throw new SyntaxException(mixinName);
+                }
+            }
+
+        }
+    }
+
+
     public ComputeBuilder url(String url) {
```

Comment:
it is a setter ? or it is a part of builder pattern ? if it is a setter method, you'd better to change the name and use void as return type, if it is builder pattern, you may need to redesign this class to involve all fields